### ISSUE TITLE
Over-zealous Module Normalization

### ISSUE THREAD

**Issue by EisenbergEffect** [frankwallis/plugin-typescript#71]
I'm not sure if this is a bug in this TypeScript plugin or if it's a bug in system.js. I'm attaching a simple Aurelia solution set up to use TypeScript and this plugin. You can simply unzip the package and view the index.html file in FireFox...or use any web server to serve it. You will see the error in the console as similar to the following:

```
Error loading file:///Users/EisenbergEffect/Desktop/kit-typescript/src/nav-bar.html.ts!html-resource-plugin
```

The module name being requested is "file:///Users/EisenbergEffect/Desktop/kit-typescript/src/nav-bar.html!html-resource-plugin" It appears, that when a call to `normalizeSync` is made against a module id that contains a plugin and a file extension, the package attempts to override the existing file extension with its own.

If this is an error in my configuration, please let me know. It appears to be a bug of some sort though. You should have everything you need to determine the source and solution for the problem in the attached zip.
[kit-typescript.zip](https://github.com/frankwallis/plugin-typescript/files/58389/kit-typescript.zip)

**Comment by frankwallis** [comment]
Firstly I don't see where html-resource-plugin is configured or comes from, is this internal to Aurelia?

In any case I think this is an issue in SystemJS, it is similar to this [issue](https://github.com/jspm/jspm-cli/issues/1308#issuecomment-163254139). These html files fall into the 'src' package and so they will be given a .ts extension regardless.
 
This does not play very well with TypeScript which does not allow you to put file extensions on imports. Currently the workarounds are to move files which don't need the ts extension appended into a separate directory so that they can use a separate package configuration.

There was a time when SystemJS would only append the default extension to files which did not already have an extension, but this meant that you could not use dots in filenames (e.g. customers.viewmodel.ts). At some point that got reverted back to where we are today.

I have [code](https://github.com/frankwallis/plugin-typescript/blob/master/src/utils.ts#L59) which (probably quite illegally) tries to make the situation better but that will not help in this case as the app.html file is not coming through plugin-typescript. Perhaps you need to add similar code to html-resource-plugin?

@guybedford - I understand that this situation will improve when the defaultJsExtensions stuff is deprecated, but it is unclear to me how that can happen in the short term - will it not break all npm packages? It seems that the options here are to (a) disallow dots in filenames or (b) force everyone to organise their source files by extension? Personally I would prefer (a).

**Comment by EisenbergEffect** [comment]
The html-resource-plugin is registered by Aurelia at runtime.

@guybedford I would not see forcing organization as a reasonable solution to this problem at all. Hopefully we can find a way to address this bug.

@frankwallis The problem is that I can't make the extension correction in my plugin because my plugin never gets called in this scenario.

**Comment by frankwallis** [comment]
using this configuration seems to get it further (normalizeSync returns http://src/nav-bar.html!html-resource-plugin) but it's still not quite booting:

```
packages: {
    "app": {
      "main": "app",
      "defaultExtension": "ts",
      "meta": {
        "nav-bar.html": {
        	"loader": "html-resource-plugin"
        },
        "*.ts": {
          "loader": "ts"
        },
        "*.js": {
          "loader": "ts"
        }
      }
    }
```

**Comment by EisenbergEffect** [comment]
I added this:

```
'*.html.ts': { loader: 'html-resource-plugin' }
```
to the meta section. I then went into my code and hacked the check for the double extension. After that I was able to get it to work...but that is not pretty.

@guybedford There needs to be some way to not have all resources from a particular package get the default extension. In my case, we had an .html extension already and a plugin, but the .ts extension was being added on and the plugin was being ignored after that point.
